var/spack/repos/builtin/packages/date/package.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Date(CMakePackage):
+    """A date and time library based on the C++11/14/17 <chrono> header"""
+
+    homepage = "https://github.com/HowardHinnant/date"
+    url = "https://github.com/HowardHinnant/date/archive/v3.0.0.zip"
+
+    version('3.0.0', sha256='ddbec664607bb6ec7dd4c7be1f9eefc3d8ce88293ffc9391486ce6ce887ec9b2')
+
+    variant('cxxstd',
+            default='17',
+            values=('11', '14', '17'),
+            description='Use the specified C++ standard when building')
+    variant('shared',
+            default=False,
+            description='Build shared instead of static libraries')
+    variant('tz', default=False, description='Build/install of TZ library')
+    variant('tzdb',
+            default='download',
+            values=('download', 'system', 'manual'),
+            description="Timezone database source (automatic download, use "
+                        "operating system's database, or manually specify)")
+
+    depends_on('cmake@3.7.0:', type='build')
+    depends_on('curl', when='+tz tzdb=download')
+
+    def cmake_args(self):
+        spec = self.spec
+        args = []
+
+        args.append('-DCMAKE_CXX_STANDARD={0}'.format(
+            spec.variants['cxxstd'].value))
+        # Require standard at configure time to guarantee the
+        # compiler supports the selected standard.
+        args.append('-DCMAKE_CXX_STANDARD_REQUIRED=ON')
+
+        args.append('-DBUILD_SHARED_LIBS={0}'.format(
+            'ON' if spec.variants['shared'].value else 'OFF'))
```

Comment:
We usually use `'+shared' in spec` for boolean variants, but I don't think this is a requirement. The CMakePackage base class also has some tools to automate this, although I haven't personally used them.

Reply:
OK, thanks for the heads up. I agree `'+shared' in spec` is nicer syntax. 

I mostly constructed this PR by reviewing similar existing recipes and hadn't really looked over CMakePackage in much detail, TBH. So I was surprised to learn that it has such nice tools for defining these cmake cli arguments. Seems like they should be more widely used.

I'll revise it to use the base class's methods. Are you aware of any exemplary cmake-based packages that I can use as reference in the future?

Reply:
`dealii` is a pretty complex CMakePackage that was recently overhauled to use those nice base class features.